osu.Desktop/Program.cs

```diff
@@ -174,8 +174,18 @@ private static bool trySendIPCMessage(IIpcHost host, string cwd, string[] args)
             return false;
         }
 
-        private static void setupVelopack()
+        private static void setupVelopack(string[] args)
         {
+            // Arguments being present indicate the user is either starting the game in a special (aka tournament) mode,
+            // or is running with pending imports via file association or otherwise.
+            //
+            // In both these scenarios, we'd hope the game does not attempt to update.
+            if (args.Length > 0)
+            {
+                Logger.Log("Handling arguments, skipping velopack setup.");
+                return;
+            }
```

Comment:
What I'd be concerned about here is that skipping `VelopackApp.Build()` / `Run()` then breaking flows like manual update checks via `Velopack.UpdateManager`, but given that [the suggestion to skip this comes from the velopack maintainer himself](https://github.com/ppy/osu/issues/34330#issuecomment-3106452427) I'm just going to assume that it's fine and not test that.

Reply:
In brief investigation, it looks like it should not have an effect on actual update flow, only startup tasks.

I did consider the other proposed approach but it's slightly more involved to ensure working correctly. The only way I can see this change going wrong is if some user decides to always send an arg to the game by accident (probably non-windows users if at all) it will never run the velopack setup. But it's logged loudly so.